ios/brave-ios/Sources/Brave/Frontend/Shields/ShieldsPanelView.swift

```diff
@@ -0,0 +1,413 @@
+      ContentBlockerManager.log.error(
+        "Failed to load favicon: \(String(describing: error))"
+      )
+    }
+  }
+}
+
+private struct ShieldSettingRow<Contents>: View where Contents: View {
+  @ViewBuilder var contents: () -> Contents
+
+  var body: some View {
+    VStack(alignment: .leading, spacing: 0) {
+      contents()
+        .frame(minHeight: 32, alignment: .center)
+        .padding(.horizontal)
+        .padding(.vertical, 4)
+      Divider()
+    }
+    .frame(maxWidth: .infinity, alignment: .leading)
+  }
+}
+
+private struct ShieldSettingsNavigationWrapper<Contents>: View where Contents: View {
+  @ViewBuilder var contents: () -> Contents
+
+  var body: some View {
+    HStack {
+      contents()
+      // Hack to showing the navigation chevron
+      Image(systemName: "chevron.right")
+        .font(.footnote)
+        .fontWeight(.medium)
+        .foregroundStyle(Color(.secondaryBraveLabel))
+    }.contentShape(Rectangle())
+  }
+}
+
+private struct ShieldSettingSectionHeader: View {
+  let title: String
+
+  var body: some View {
+    VStack(alignment: .leading, spacing: 8) {
+      Text(title)
+        .font(.footnote)
+        .foregroundStyle(Color(.secondaryBraveLabel))
+        .textCase(.uppercase)
+        .padding(.horizontal)
+        .padding(.top, 8)
+        .padding(.bottom, 0)
+      Divider()
+    }.padding(.top, 8)
+  }
+}
+
+class ShieldsPanelViewController: UIHostingController<ShieldsPanelView>, PopoverContentComponent {
+  private let shieldsPanelView: ShieldsPanelView
+
+  init(url: URL, tab: Tab, domain: Domain, callback: @escaping (ShieldsPanelView.Action) -> Void) {
+    let shieldsPanelView = ShieldsPanelView(
+      url: url,
+      tab: tab,
+      domain: domain,
+      callback: callback
+    )
+    self.shieldsPanelView = shieldsPanelView
+    super.init(rootView: shieldsPanelView)
+  }
+
+  @MainActor required dynamic init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    sizingOptions = .preferredContentSize
+  }
+}
+
+private struct ShieldsPanelDisclosureStyle: DisclosureGroupStyle {
+  func makeBody(configuration: Configuration) -> some View {
+    VStack(alignment: .leading, spacing: 0) {
+      Button {
+        withAnimation {
+          configuration.isExpanded.toggle()
+        }
+      } label: {
+        VStack(spacing: 0) {
+          Divider()
+          HStack {
+            configuration.label
+            Group {
+              if configuration.isExpanded {
+                Image(systemName: "chevron.down")
+              } else {
+                Image(systemName: "chevron.right")
+              }
+            }
+            .foregroundStyle(Color(.secondaryBraveLabel))
+            .font(.body)
+          }
+          .frame(maxWidth: .infinity, alignment: .center)
+          .padding()
+          .contentShape(Rectangle())
+        }
+      }
+      .padding(0)
```

Comment:
Why do you need `.padding(0)`?

Reply:
so i can stretch certain elements, such as dividers and the headers left and right fully as is in the original screen.